internal/credentials/credentials.go

```diff
@@ -86,7 +100,7 @@ type fileCredential struct {
 	Key string `toml:"key"`
 }
 
-func (cs *CredentialsService) FileCredentialRecordFactory() (*CredentialRecord, error) {
+func (cs *credentialsService) FileCredentialRecordFactory() (*CredentialRecord, error) {
```

Comment:
I'm not following why you changed this to an unexported struct? 
Couldn't you have just changed the existing CredentialsService to have a logger? 

Perhaps it would explain your statement from this morning of why you had to break it out to a separate logger?

Reply:
This change is to prevent panics, since `credentialsService` will call a logger instance internally, making the struct private and the interface and constructor public, forces consumers of this service to pass a logger. E.g:

```go
// The following would bring a panic, program will compile with no issues but with the risk of runtime errors
// "runtime error: invalid memory address or nil pointer dereference"
cs := credentialsService{}
cs.Get(someGuid) // <- If logger is called for any reason within this method, program will panic
```

```go
// Using the private struct but a public constructor prevents that panic
cs := NewCredentialsService(lgr) // <- constructor expects and forces us to pass a logger for the resulting struct
cs.Get(someGuid) // Won't panic, no runtime errors
```